internal/server/config.go

```diff
@@ -46,10 +46,58 @@ type KeyProvider struct {
 	Config interface{} // contains secret-provider-specific config
 }
 
-type nativeSecretProviderConfig struct {
+type nativeKeyProviderConfig struct {
 	SecretProvider string `mapstructure:"secretProvider"`
 }
 
+type AWSConfig struct {
+	Endpoint        string `mapstructure:"endpoint" validate:"required"`
+	Region          string `mapstructure:"region" validate:"required"`
+	AccessKeyID     string `mapstructure:"accessKeyID" validate:"required"`
+	SecretAccessKey string `mapstructure:"secretAccessKey" validate:"required"`
+}
+
+type AWSKMSConfig struct {
+	AWSConfig `mapstructure:",squash"`
+
+	EncryptionAlgorithm string `mapstructure:"encryptionAlgorithm"`
+	// aws tags?
+}
+
+type AWSSecretsManagerConfig struct {
+	AWSConfig `mapstructure:",squash"`
+
+	UseSecretMaps bool `mapstructure:"useSecretMaps"`
```

Comment:
I believe this `UseSecretMaps` field is never used. We now pass it to the secrets package, but I don't think the secrets package uses it either.  Might be a good opportunity to remove it until we need it.